lib/frontend/templates/subtemplates/render_sentence.rs.html

```diff
@@ -0,0 +1,25 @@
+@use japanese::furigana::SentencePartRef;
+@type MyIterator<'a> = Vec<SentencePartRef<'a>>;
```

Comment:
Why are you using a Vec? An iterator would be more storage efficient since furigana is in most cases an iterator.

Reply:
I couldn't figure out how to sneak a trait into the ructe syntax -- I'm not too advanced at Rust but I think I understand that how you would normally do this is something like 
```
func myFunc<'a, I>(iter: I)
where
    I: Iterator<Item = SentencePartRef<'a>>
{...
```

Which I couldn't figure out how to do with ructe

Reply:
`@type MyIterator<'a> = dyn Iterator<Item = SentencePartRef<'a>>;` Should work

Reply:
not fixed, added todo